lib/streamlit/commands/query_params.py

```diff
@@ -142,3 +143,46 @@ def _ensure_no_embed_params(
         separator = "&" if current_embed_params else ""
         return separator.join([query_string, current_embed_params])
     return current_embed_params
+
+
+@dataclass
+class QueryParams:
+    """A dict-like representation of query params that generally behaves like st.session.state.
```

Comment:
I feel like `st.session_state` is probably not the best comparison to make here. Can't we just say that it's a dict-like representation and leave it at that?

Reply:
Yep, I just copied it from the query_params spec so I can change that.